crates/prover-services/src/parallel/mod.rs

```diff
@@ -113,7 +113,7 @@ where
         )
     }
 
-    async fn prove_all(&self, proof_queue: Vec<ProofData>) -> Vec<Proof> {
+    async fn prove_all(&self, elf: Vec<u8>, proof_queue: Vec<ProofData>) -> Vec<Proof> {
```

Comment:
what if different proofs in the queue require different elfs

Reply:
The batch prover should break up commitments and therefore we'll have different inputs in which the inputs are related to non-forking commitments.